src/main/java/example/WebhookController.java

```diff
@@ -0,0 +1,87 @@
+package example;
+
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.google.gson.Gson;
+
+import models.submissions.Webhooks.CompletedWebhookModel;
+import models.submissions.Webhooks.CreditsCheckedWebhookModel;
+import models.submissions.Webhooks.ErrorWebhookModel;
+import models.submissions.Webhooks.IndexedWebhookModel;
+
+/**
+ * Controller to handle webhook callbacks from the external service.
+ * 
+ * Provides separate endpoints for each webhook event status:
+ * - /webhook/completed
+ * - /webhook/error
+ * - /webhook/indexed
+ * - /webhook/creditsChecked
+ * 
+ * Each endpoint accepts a POST request with a JSON payload corresponding to the
+ * event type.
+ */
+@RestController
+public class WebhookController {
```

Comment:
add above the class and below the @RestController
this: @RequestMapping("/webhook")
to reduce duplicated use of the word "/webhook" in each endpoint.